src/index.html

```diff
@@ -1,61 +1,69 @@
+  <script>
+    var webAuth = new auth0.WebAuth({
+      domain: '[YOUR_AUTH0_DOMAIN]', // e.g., you.auth0.com
+      clientID: '[YOUR_AUTH0_CLIENT_ID]',
+      scope: 'email',
+      responseType: 'token',
+      redirectUri: 'http://localhost:8888'
+    });
+    var storedProfile = localStorage.getItem('profile');
+    var storedToken = localStorage.getItem('token');
+    var authData = storedProfile && storedToken ? { profile: JSON.parse(storedProfile), token: storedToken } : null;
+    var elmApp = Elm.Main.fullscreen(authData);
 
-        // Log out of Auth0 subscription
-        elmApp.ports.auth0logout.subscribe(function(opts) {
-            localStorage.removeItem('profile');
-            localStorage.removeItem('token');
-        });
+    // Auth0 authorize subscription
+    elmApp.ports.auth0authorize.subscribe(function(opts) {
+      webAuth.authorize();
+    });
 
-        // Listening for the authenticated event
-        lock.on("authenticated", function(authResult) {
-            // Use the token in authResult to getProfile() and save it to localStorage
-            lock.getProfile(authResult.idToken, function(err, profile) {
-                var result = { err: null, ok: null };
-                var token = authResult.idToken;
+    // Log out of Auth0 subscription
+    elmApp.ports.auth0logout.subscribe(function(opts) {
+      localStorage.removeItem('profile');
+      localStorage.removeItem('token');
+    });
 
-                if (!err) {
-                    result.ok = { profile: profile, token: token };
-                    localStorage.setItem('profile', JSON.stringify(profile));
-                    localStorage.setItem('token', token);
-                } else {
-                    result.err = err.details;
+    // Watching for hash after redirect
+    webAuth.parseHash({ hash: window.location.hash }, function(err, authResult) {
+      if (err) {
+        return console.error(err);
+      }
+      if (authResult) {
+        webAuth.client.userInfo(authResult.accessToken, function(err, profile) {
+          var result = { err: null, ok: null };
+          var token = authResult.accessToken;
 
-                    // Ensure that optional fields are on the object
-                    result.err.name = result.err.name ? result.err.name : null;
-                    result.err.code = result.err.code ? result.err.code : null;
-                    result.err.statusCode = result.err.statusCode ? result.err.statusCode : null;
-                }
-                elmApp.ports.auth0authResult.send(result);
-            });
+          if (err) {
+            result.err = err.details;
+            // Ensure that optional fields are on the object
+            result.err.name = result.err.name ? result.err.name : null;
+            result.err.code = result.err.code ? result.err.code : null;
+            result.err.statusCode = result.err.statusCode ? result.err.statusCode : null;
+          }
+          if (authResult) {
```

Comment:
I think this should be `else` or `if (profile)`. `if (authResult)` makes no sense here since `authResult` is always `true` at this point.